src/main.rs

```diff
@@ -66,7 +67,16 @@ fn main() {
     let sha = if args.len() > 1 {
         args[1].to_owned()
     } else {
-        format!("{:8x}", rand::random::<usize>())
+        // no args, check stdin
+        let stdin = io::stdin();
+        let mut reader = stdin.lock();
+        let mut buffer = String::new();
+        reader.read_to_string(&mut buffer);
```

Comment:
This actually is slightly flawed. We should be able to read each line and then output the result. Right now we hang with this:
```
yes | cargo run
```

We may also want to fix this in grusp now that I think about it.